R/epidist_utils.R

```diff
@@ -377,50 +383,45 @@ create_epidist_summary_stats <- function(mean = NA_real_,
 #' )
 create_epidist_citation <- function(author = NA_character_,
                                     year = NA_integer_,
-                                    PMID = NA_character_,
-                                    DOI = NA_character_) {
+                                    title = NA_character_,
+                                    journal = NA_character_,
+                                    DOI = NA_character_,
+                                    PMID = NA_character_) {
   # check input
   checkmate::assert_character(author)
   checkmate::assert_number(year, na.ok = TRUE)
+  checkmate::assert_character(title)
+  checkmate::assert_character(journal)
   checkmate::assert_number(PMID, na.ok = TRUE)
   checkmate::assert_character(DOI)
 
-  if (is.na(author) || is.na(year) || is.na(DOI)) {
+  if (is.na(author) || is.na(year) || is.na(journal) || is.na(title)) {
     message(
-      "Citation cannot be created as either author, year or DOI is missing"
+      "Citation cannot be created as author, year, journal or title is missing"
     )
-    return("No citation available")
+    return(utils::bibentry(bibtype = "Misc", title = "No citation"))
   }
 
-  # change author formatting if multiple authors or et al
-  author <- gsub(
-    pattern = "_",
-    replacement = " ",
-    x = author,
-    fixed = TRUE
-  )
-  author <- gsub(
-    pattern = "etal",
-    replacement = "et al.",
-    x = author,
-    fixed = TRUE
-  )
-
-  # check if study has two authors and if so insert ampersand
-  num_authors <- length(unlist(strsplit(x = author, split = " ", fixed = TRUE)))
-  if (identical(num_authors, 2L)) {
-    author <- gsub(pattern = " ", replacement = " & ", x = author, fixed = TRUE)
+  if (!inherits(author, "person")) {
+    # imperfect solution as library currently only has first author
+    author_names <- unlist(strsplit(x = author, split = "_", fixed = TRUE))
```

Comment:
```suggestion
    author_names <- strsplit(x = author, split = "_", fixed = TRUE)
```

`unlist()` is not necessary since `lapply()` is used right after.